### ISSUE TITLE
editing a page sometimes changes date of page to '1 January 0001, at 00:00'

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#438]
I have been editing pages in enwikisource and noticed that some of the pages I had edited had a date of '1 January 0001, at 00:00'
So I then chose another page at random and edited it - this page retained its original date.

I have discovered that the issue arises when editing is done in two stages.
That is, first a 'Show preview' followed by 'Save page'
The 'Show preview' is not maintaining the date (and makes one up) - if it is going to do that why not today's date and time

**Comment by desb42** [comment]
And another small thing. Whilst investigating this, my edit changes tended to be adding (or removing) spaces
I noticed that there is a routine that updates the page_touched field and was wondering if it would be possible to change page_len at the same time (as no mater how many spaces I added the page_len never changed)

**Comment by gnosygnu** [comment]
> That is, first a 'Show preview' followed by 'Save page'
> The 'Show preview' is not maintaining the date (and makes one up) - if it is going to do that why not today's date and time

Yeah, that's  a bug. It's probably also related to #435 b/c Show Preview does some gymnastics to render the wikitext while still allowing edit

> And another small thing. Whilst investigating this, my edit changes tended to be adding (or removing) spaces

I don't think it should be adding spaces. MediaWiki has behavior to automatrically trim all white-space from the end upon edit. XOWA specifically implements this behavior (I forget why, but it caused some weird bug without it). I'm assuming you're seeing the auto-trim-at-end

> I noticed that there is a routine that updates the page_touched field and was wondering if it would be possible to change page_len at the same time (as no mater how many spaces I added the page_len never changed)

Nope. XOWA doesn't do anything to force the article to maintain the same page_len. However, you are probably discovering another bug wherein the page_len is not being updated.
